File: CSharp/ByRound/1531/1531_A_color.cs
// Problem: CF 1531 A - Зингер | color
// https://codeforces.com/contest/1531/problem/A

﻿/*
 * Код обрабатывает последовательность сообщений для управления цветом подсветки в куполе Дома Зингера.
 * Алгоритмы/Techniques: Симуляция состояния с использованием флага блокировки и переменной последнего цвета.
 * 
 * Временная сложность: O(n), где n — количество сообщений.
 * Пространственная сложность: O(1), так как используются только постоянные переменные.
 */

using System;
namespace ConsoleApp1 {
class Program {
  static void Main(string[] args) {
    bool isLocked = false; // Флаг, indicating если изменение цвета заблокировано
    string lastColor = "blue"; // Последний установленный цвет (начинается с "blue")
    int n = int.Parse(Console.ReadLine()); // Количество сообщений
    
    for (; n > 0; n--) {
      string input = Console.ReadLine(); // Чтение следующего сообщения
      
      if (input == "lock")
        isLocked = true; // Блокировка изменения цвета
      else if (input == "unlock")
        isLocked = false; // Разблокировка изменения цвета
      else if (!isLocked)
        lastColor = input; // Установка нового цвета, если не заблокировано
    }
    
    Console.WriteLine(lastColor); // Вывод конечного цвета
  }
}
}


// https://github.com/VaHiX/codeForces/

File: CPP/ByRound/1531/1531_A_color.cpp
// Problem: CF 1531 A - Зингер | color
// https://codeforces.com/contest/1531/problem/A

#include <iostream>
#include <string>

using namespace std;

/*
 * Code purpose: This program simulates the color changing process of the Zinger dome based on a series of commands.
 *               It tracks the current color and whether color changes are locked or not.
 * 
 * Algorithms/Techniques: 
 *   - Linear simulation of command processing
 *   - State tracking using a flag (fl) for lock state
 *   - String comparison for command interpretation
 * 
 * Time Complexity: O(n), where n is the number of messages. Each message is processed exactly once.
 * Space Complexity: O(1), only a constant amount of extra space is used regardless of input size.
 */
int main() {
  int n, fl = 0;  // fl = 0 means unlocked, fl = 1 means locked
  string s = "blue";  // Initial color is blue
  cin >> n;
  for (int i = 1; i <= n; ++i) {
    string st;
    cin >> st;
    if (st == "lock")
      fl = 1;  // Lock the color change
    else if (st == "unlock")
      fl = 0;  // Unlock the color change
    else if (fl == 0)  // If not locked, change the color
      s = st;
  }
  cout << s;
}


// https://github.com/VaHiX/CodeForces/